### PR TITLE
TranslationDeepl init

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: config/packages/translation_deepl.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+parameters:
+    deepl_auth_key: '%env(DEEPL_AUTH_KEY)%'
```
Comment (BastekBielawski) [inline config/packages/translation_deepl.yml]: Should be default value if env not exists

File: module/translation-deepl/composer.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "doctrine/dbal": "^2.9",
+        "ergonode/core": "^0.2.0",
+        "symfony/http-foundation": "^4.2"
+    },
+    "autoload": {
+        "psr-4": {
+            "Ergonode\\TranslationDeepl\\": "src/"
+        }
+    }
+}
```
Comment (BastekBielawski) [inline module/translation-deepl/composer.json]: Verify, after all with @rprzedzik

File: module/translation-deepl/src/Application/Controller/Api/TranslationDeeplController.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    private $repository;
+
+    public function __construct(MessageBusInterface $messageBus, TranslationDeeplRepositoryInterface $repository)
+    {
+
+        $this->messageBus = $messageBus;
+        $this->repository = $repository;
+    }
+
+    public function getTranslation(Request $request): Response
```
Comment (BastekBielawski) [inline module/translation-deepl/src/Application/Controller/Api/TranslationDeeplController.php]: Route declaration with API doc

File: module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+declare(strict_types=1);
+
+
+namespace Ergonode\TranslationDeepl\Application\Model\Form\Type;
+
+use Symfony\Component\Validator\Constraints as Assert;
+
+
+class TranslationDeeplConfigurationTypeModel
```
Comment (BastekBielawski) [inline module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php]: DeeplConfigurationModel

File: module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+{
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank(message="Target Langauge is required")
+     * @Assert\Length(
```
Comment (BastekBielawski) [inline module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php]: Add `)`

File: module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+     *      min = 2,
+     *      max = 2,
+     */
+    // TODO: Add Custom Validator
+    public $targetLang;
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank(message="Source Language is required")
+     * @Assert\Length(
```
Comment (BastekBielawski) [inline module/translation-deepl/src/Application/Model/Form/Type/TranslationDeeplConfigurationTypeModel.php]: Add `)`

File: module/translation-deepl/src/Infrastructure/Provider/TranslationDeeplProviderInterface.php [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+declare(strict_types = 1);
+
+namespace Ergonode\TranslationDeepl\Infrastructure\Provider;
+
+
+use Scn\DeeplApiConnector\Model\TranslationConfig;
+
+interface TranslationDeeplProviderInterface
+{
+    public function provide(TranslationConfig $translation);
```
Comment (BastekBielawski) [inline module/translation-deepl/src/Infrastructure/Provider/TranslationDeeplProviderInterface.php]: Return type

File: module/translation-deepl/src/Infrastructure/Provider/Decorator/CacheTranslationDeeplProviderDecorator.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\TranslationDeepl\Infrastructure\Provider\Decorator;
+
+
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProviderInterface;
+
+class CacheTranslationDeeplProviderDecorator implements TranslationDeeplProviderInterface
```
Comment (rprzedzik) [inline module/translation-deepl/src/Infrastructure/Provider/Decorator/CacheTranslationDeeplProviderDecorator.php]: Add sample dbal base cache with uuid key, cache data and life time.

Comment (rprzedzik) [reply module/translation-deepl/src/Infrastructure/Provider/Decorator/CacheTranslationDeeplProviderDecorator.php]: Uidd key should be generate from string something like this:

```
$ns = 'uniqu uuid for cache';
$name = sprintf('%s_%s_%s' ,$sourceLanguage, $desonationLangauge, $phaseToTranslate);
$uuid = Uuid::Uuid5($namespace, $name);
```

File: module/translation-deepl/src/Persistence/Dbal/Repository/DbalTranslationDeeplRepository.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+use Ergonode\EventSourcing\Domain\AbstractAggregateRoot;
+use Ergonode\EventSourcing\Infrastructure\DomainEventDispatcherInterface;
+use Ergonode\EventSourcing\Infrastructure\DomainEventStoreInterface;
+use Ergonode\TranslationDeepl\Domain\Entity\TranslationDeepl;
+use Ergonode\TranslationDeepl\Domain\Entity\TranslationDeeplId;
+use Ergonode\TranslationDeepl\Domain\Repository\TranslationDeeplRepositoryInterface;
+
+/**
+ */
+class DbalTranslationDeeplRepository implements TranslationDeeplRepositoryInterface
```
Comment (rprzedzik) [inline module/translation-deepl/src/Persistence/Dbal/Repository/DbalTranslationDeeplRepository.php]: Unnecessary class

**Pull request closed without merging**